src/components/tag/tag.icon.tsx

```diff
@@ -0,0 +1,52 @@
+import { QuillSvgProps } from '@deriv/quill-icons'
+import {
+  BaseTagIconProps,
+  BaseTagProps,
+  BaseTagSizeProps,
+  TagIconFillCVA,
+  TagIconSizes,
+  TagIcons,
+} from './tag.classnames'
+
+export interface TagIconProps extends BaseTagSizeProps, BaseTagIconProps {
+  icon?: React.ForwardRefExoticComponent<Omit<QuillSvgProps, 'ref'>>
+  className?: string
+  isBold: BaseTagProps['isBold']
+}
+
+export const TagIcon = ({
+  icon: Icon,
+  size,
+  className,
+  isBold,
+  colorStyle,
+}: TagIconProps) => {
+  let IconComponent:
+    | React.ForwardRefExoticComponent<Omit<QuillSvgProps, 'ref'>>
+    | null
+    | undefined = null
+  if (colorStyle === 'custom') {
+    IconComponent = Icon
+  } else {
+    IconComponent =
+      TagIcons[colorStyle ?? 'success'][isBold ? 'bold' : 'regular']
+  }
+  return (
+    <>
+      {Icon && colorStyle === 'custom' ? (
+        <Icon className={className} {...TagIconSizes[size ?? 'md']} />
```

Comment:
@aizad-deriv if this optional chaining is because of the `CVA null` behavior then we can make use of this `ExcludeNull<T, 'key1' | 'key2' | 'key3'>` type. It's already in the main branch. 🙏